src/idd.ml

```diff
@@ -190,6 +192,18 @@ let star mgr (d0:t) =
 let subseteq mgr (d0:t) (d1:t) =
   equal (union mgr d0 d1) d1
 
+
+
+let idd_from_dd mgr (dd:Dd.t) =
+  let idd_from_dd' mgr dd =
+    match dd with
+    | Dd.True -> ident
+    | Dd.False -> empty
+    | Dd.Branch { var; hi; lo; } -> branch mgr var hi lo
+  in
+  idd_from_dd' (manager ~d_mgr:mgr ()) dd
```

Comment:
This creates a new manager and then immediately discards it. As a result, the id of the returned diagram is actually arbitrary (undefined).

If you really just want to cast, you can simply make this the identity function:
```
let idd_from_dd (dd : Dd.t) : t = dd
```